### PR TITLE
begin adding functionality

### PR DESCRIPTION
**Pull request by baileyjarvis2814**
The purpose of this pr is to focus on the beginning elements of functionality. I was failing to properly return the data I fetched via my api calls as arrays I could easily manipulate. Because of that I couldn't get the proper booking information to show up for the specific customer data. Now I am able to see the bookings and total amount spent for any given logged in customer. A customer without bookings or money spent will see no bookings and a total amount spent of $0.00.

Extra information, there was also more testing being done to try and continue tdd and get a breakthrough on generating cards for the booking information to be displayed on.

### CODE REVIEW

File: dist/index.html
Comment (moth-dust) [inline dist/index.html:1]: pseudo coding placeholders for html was a good idea!

File: src/booking.js
Comment (moth-dust) [inline src/booking.js:1]: These are good and clear input validations!

File: src/customer.js [hunk 1]
```diff
@@ -11,18 +24,29 @@ export const createCustomer = (customer, bookings) => {
...
     name: customer.name,
-    getCustomerBookings: () => getCustomerBookings(customer.id, bookings),
-    getTotalCost(bookings, rooms) {
-      const customerBookings = getCustomerBookings(customer.id, bookings);
-      const customerRoomNumbers = customerBookings.map(booking => booking.roomNumber);
-      const customerRooms = rooms.filter(room => customerRoomNumbers.includes(room.number));
+    getCustomerBookings: () => customerBookings,
+    getTotalCost: () => calculateTotalCost(customerBookings, rooms),
+    getRoomToBook: (date, roomNumber) => {
+      if (date.includes('-')) {
```
Comment (moth-dust) [inline src/customer.js:35]: excellent closure here.

File: src/room.js [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
+      }).join(' ');
+    },
+    getBedSize() {
+      return this.bedSize[0].toUpperCase() + this.bedSize.substring(1);
+    }
   };
+  
+
+  return room;
 };
```
Comment (moth-dust) [inline src/room.js:34]: I like the way you are itemizing concerns, very clean modifiable coding.

**Review by moth-dust — APPROVED** [review]

**Review by KojinKuro — COMMENTED** [review]
My main feedback would be to include instructions on how to replicate the results you tested for (getting the global booking, customers, and rooms working) It's not automatically clear given the website is still in early phases. I could not figure out how to get the money spent for a user or logging in. I did figure out that you `console.log()` the global data vars by reading your code, but might take a while for someone new to the codebase.

You should be more clear in describing what your PR is about. You mention "adding functionality" as the main thing for the PR, but most of the changes were refactoring. I would also break down what adding functionality means as it is vague.

Also why the tests are being skipped now?

**Pull request merged**